tests/unit/conftest.py

```diff
@@ -7,3 +10,26 @@
 @pytest.fixture(name='root_logger')
 def fixture_root_logger(caplog: _pytest.logging.LogCaptureFixture) -> Logger:
     return Logger.create(verbose=0, debug=0, quiet=False)
+
+
+@pytest.fixture(scope='module')
+def source_dir():
```

Comment:
Wouldn't be pytest's `tmp_path` fixture better?

Reply:
Interesting fixture, never seen it used, I like when I know what the code is doing exactly and cleaning up after is kind of important part. I'm not sure if it would help anything as most of this fixture would have to stay anyway.

Reply:
Yeah, I'd recommend using Pytest's existing `tmp_path` ficture as a basis for your work, they can be combined - creating the directory directly under `/tmp` might seem harmless, but quite often, pytest is told what its "tmpdir" should be (`--basetemp`), and this action would disobey such an order.

```python
@pytest.fixture(scope='module')
def source_dir(tmp_path):
    """ Create dummy directory structure and remove it after tests """
    source_location = tmp_path
    (source_location / 'library').mkdir(parents=True)
    (source_location / 'lib_folder').mkdir()
    (source_location / 'tests').mkdir()
    for num in range(10):
        test_path = source_location / f'tests/bz{num}'
        test_path.mkdir()
        (test_path / 'runtests.sh').touch()
    yield source_location
    shutil.rmtree(source_location, ignore_errors=True)
```

Reply:
One more thing: while it should be safe to remove the directory, pytest does not care about it that much, it's just a space in filesystem allocated for your scope level, I would advise against it. Unless it's a big chunk of space, it's often nicer to keep the files around for future investigators. Imagine the test fails - the way the fixture is writte, it'd remove the files and directories, leaving me no chance for investigation of this part of the test. Was it something I broke in the code, or was it dummy files created for the test? Can't tell, files are gone :) Unless there's a kernel source unzipped in those directories, please, keep them around for less luckier days.

I appreciate the habit of cleaning up your toys when you're done with them, leaving no trail in `/tmp`, but that's exactly what `tmp_path` fixture is for - a dedicated space for that particular pytest run, that can be removed entirely when noone is interested in its content. Basically the decision whether to remove the content is owned not by the test, but by whoever ran the test as they have more info - will the files be needed for anything when pytest finishes?

Reply:
WRT "Interesting fixture, never seen it used" - look for `tmpdir` under `tests/unit`. It's an older fixture, does the very same thing, but returns different class of objects (`py.local` vs `Path`). It's not an uncommon fixture to use, really :)

Reply:
allright, no problem changing it to tmp_path
the 'never seen it used' was referring to my old team where we used pytest quite a lot, but the situation there (for generating paths/names) was a little different, thanks for all the insights

Reply:
Our (read "in Testing Farm") history is different, we use it heavily: our tests often run in containers, and we point pytest to a volume mounted into the container. Containers are thrown away, so no mess left around when pytest finishes, but files outside of the location we pass to pytest would be lost for the host. That's often a Jenkins build, and artifacts outside of a given directory cannot be stored as build artifacts.

So, I come from environment where theese kinds of separation and constraints are crucial, and I actually like it as it makes life slightly easier for the test: here's your directory - no, please, really, you don't have to invent your own, or think about whether it should be in `/tmp` or `/var/tmp`, seriously, use this one, and focus on testing  :)